src/Raven.Server/Commercial/LicenseStorage.cs

```diff
@@ -0,0 +1,172 @@
+
+                Slice keyAsSlice;
+                TableValueReader infoTvr;
+                using (Slice.From(tx.InnerTransaction.Allocator, FirstServerStartDateKey, out keyAsSlice))
+                {
+                    infoTvr = table.ReadByKey(keyAsSlice);
+                }
+                //It seems like the database was shutdown rudly and never wrote it stats onto the disk
+                if (infoTvr == null)
+                    return null;
+
+                var firstServerStartDateJson = Read(context, infoTvr);
+                DateTime result;
+                if (firstServerStartDateJson.TryGet(FirstServerStartDateKey, out result))
+                    return result;
+
+                return null;
+            }
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private unsafe BlittableJsonReaderObject Read(JsonOperationContext context, TableValueReader reader)
+        {
+            int size;
+            var ptr = reader.Read(LicenseInfoSchema.LicenseTable.JsonIndex, out size);
+            return new BlittableJsonReaderObject(ptr, size, context);
+        }
+
+        public unsafe void SaveLicense(DynamicJsonValue license)
+        {
+            TransactionOperationContext context;
+            using (_contextPool.AllocateOperationContext(out context))
+            using (var tx = context.OpenWriteTransaction())
+            {
+                var table = tx.InnerTransaction.OpenTable(_licenseStorageSchema, LicenseInfoSchema.LicenseTree);
+
+                using (var id = context.GetLazyString(LicenseStoargeKey))
+                using (var json = context.ReadObject(license, LicenseStoargeKey,
+                    BlittableJsonDocumentBuilder.UsageMode.ToDisk))
+                {
+                    var tvb = new TableValueBuilder
+                    {
+                        {id.Buffer, id.Size},
+                        {json.BasePointer, json.Size}
+                    };
+
+                    table.Set(tvb);
+                }
+                tx.Commit();
+            }
+        }
+
+        public License LoadLicense()
+        {
+            TransactionOperationContext context;
+            using (_contextPool.AllocateOperationContext(out context))
+            using (var tx = context.OpenReadTransaction())
+            {
+                var table = tx.InnerTransaction.OpenTable(_licenseStorageSchema, LicenseInfoSchema.LicenseTree);
+
+                TableValueReader reader;
+                Slice key;
+                using (Slice.From(context.Allocator, LicenseStoargeKey, out key))
+                {
+                    reader = table.ReadByKey(key);
+                }
+                if (reader == null)
+                    return null;
+
+                var licenseJson = Read(context, reader);
+                return JsonConvert.DeserializeObject<License>(licenseJson.ToString());
```

Comment:
Why?
We already have a blittable document, and then we can use the `JsonServerDeserialization` there.